lib/vertica/connection.rb

```diff
@@ -2,56 +2,43 @@
 
 class Vertica::Connection
 
-  attr_reader :notices, :transaction_status, :backend_pid, :backend_key, :parameters, :notice_handler, :session_id
+  attr_reader :transaction_status, :backend_pid, :backend_key, :parameters, :notice_handler, :session_id
 
-  attr_accessor :row_style, :debug, :options
+  attr_reader :options
 
   def self.cancel(existing_conn)
     existing_conn.cancel
   end
 
   # Opens a connectio the a Vertica server
   # @param [Hash] options The connection options to use.
-  def initialize(options = {})
-    reset_values
+  def initialize(host: nil, port: 5433, username: nil, password: nil, database: nil, interruptable: false, ssl: nil, read_timeout: 600,row_style: :hash, debug: false, role: nil, search_path: nil, timezone: nil, skip_startup: false)
+    reset_state
     @notice_handler = nil
 
-    @options = {}
-    @debug = false
-
-    options.each { |key, value| @options[key.to_s.to_sym] = value if value}
-
-    @options[:port] ||= 5433
-    @options[:read_timeout] ||= 600
-
-    @row_style = @options[:row_style] ? @options[:row_style] : :hash
-    boot_connection unless options[:skip_startup]
+    @options = {
+      host: host,
+      port: port.to_i,
+      username: username,
```

Comment:
- I always used `username` myself, and it is my preference because it is the standard that ActiveRecord and `database.yml` use. Having said that, adding a fallback to using `user` is fine.
- The most commonly used APIs are still the same, but a lot of the internal APIs that were exposed before have significantly changed. Because of that, a new major version is in order I think.